src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -323,15 +317,32 @@ def _run_task(self, ahs_program):
         return task
 
     def _validate_pulses(self, pulses):
-
-        # ToDo: allow local drive
-
-        if len(pulses) > 1:
-            raise NotImplementedError(
-                f"Multiple pulses in a Rydberg Hamiltonian are not currently supported. "
-                f"Recieved {len(pulses)} pulses.")
-
-        if pulses[0].wires != self.wires:
-            raise NotImplementedError(
-                f"Only global drive is currently supported. Found drive defined for subset "
-                f"{[pulses[0].wires]} of all wires [{self.wires}]")
\ No newline at end of file
+        # Iterate through pulses to find global drive
+        max_wires = 0
+        for i, pulse in enumerate(pulses):
+            if len(pulse.wires) > max_wires:
+                max_wires = len(pulse.wires)
+                global_index = i
```

Comment:
Could you also look for `len(pulse.wires) == len(self.wires)` and make sure there is only one of them? It reads a bit complicated, but I guess this works, just not if one puts multiple global pulses. Is this checked somewhere else?

Reply:
Agreed, we want to raise an error here if there is more than one global pulse (here I think we don't catch that at all, if it encounters a second pulse with just as many wires it just moves on).

If we pass the full operator to this function instead of only the pulses, we could access `op.wires`, and select the global drive(s) as the one(s) acting on all the wires in `op.wires`. I think that would make this simpler.